bin/github-review.sh

```diff
@@ -69,68 +69,8 @@ fi
 
 echo "Checking changes to assets files"
 fileLineChangeJSON=$(gh pr view $pr --json files) # This line fetches the JSON output of the files changed in the PR
-isHighestChainID=0
-echo "$fileLineChangeJSON" | jq -r '.files[] | "\(.path) \(.additions) \(.deletions)"' | while read -r line; do
-    path=$(echo $line | cut -d ' ' -f1)
-    additions=$(echo $line | cut -d ' ' -f2)
-    deletions=$(echo $line | cut -d ' ' -f3)
-
-    # Now you can perform checks on $additions and $deletions as per your requirements
-    if [[ ("$additions" == 2 && "$deletions" == 1) ]]; then
-        isHighestChainID=1
-    elif [[ ("$additions" != 1 || "$deletions" != 0) ]]; then
-        echo "ERROR: $path has invalid changes" 1>&2
-        exit 1
-    fi
-done
-
-if [[ $isHighestChainID == 1 ]]; then
-    echo "Edge case when adding chain with the highest chain id number"
-    diffPatchSeparated=($(gh pr diff $pr | grep -E '^[+-] '))
-    # Adding three elements at a time together to compare from the output array
-    diffPatch=()
-    for ((i=0; i<${#diffPatchSeparated[@]}; i+=3)); do
-        diffPatch+=("${diffPatchSeparated[i]} ${diffPatchSeparated[i+1]} ${diffPatchSeparated[i+2]}")
-    done
-
-    # Initialize a counter
-    counter=1
-    # Initialize a flag to indicate if the previous pattern was correct
-    pattern_correct=true
-    # Read input line by line from the array
-    for line in "${diffPatch[@]}"; do
-        # Determine the line type based on the counter
-        case $((counter % 3)) in
-            1) # First line should start with '-'
-                if [[ $line != -* ]]; then
-                    pattern_correct=false
-                fi
-                # Store the first line to compare with the second line
-                first_line="$line"
-                ;;
-            2) # Second line should be the first line with a comma
-                expected_line="+${first_line:1},"
-                if [[ $line != $expected_line ]]; then
-                    pattern_correct=false
-                fi
-                ;;
-            0) # Third line should start with '+'
-                if [[ $line != +* ]]; then
-                    pattern_correct=false
-                fi
-                # Check the pattern for the set of three lines
-                if [ "$pattern_correct" = false ]; then
-                    echo "Unknown lines added or removed" 1>&2
-                    exit 1
-                fi
-                # Reset the pattern_correct flag for the next set of lines
-                pattern_correct=true
-                ;;
-        esac
-        # Increment the counter
-        ((counter++))
-    done
-fi
+diffPatchSeparated=$(gh pr diff $pr | grep -E '^[+-] ' | base64) # Sending multiple lines as a string through base64 encoding
```

Comment:
Why encode it as base64?

Reply:
Ah, is this because environment variables are expected to be single line?

Reply:
Yes

Reply:
I think you can quickly run into character limits with this. Personally, I think the "correct" approach would be to either:

- use the `ChildProcess` API in the script to compute the diff
- use files. You can even make the script read the diff over STDIN to the node process. An easy approach would be to either:
```
gh pr diff $pr > .temp-github-review.diff
npm run verifyChanges -s -- --fileLineChange "$fileLineChangeJSON" --diffPatch .temp-github-review.diff  --verbose
```

I'm also not a fan of processing the input before passing it to the script instead of just having the script filter lines its interested in (i.e. do the `grep -E '^[+- ] '` in JavaScript). My rationale is that the script then accepts a standardized format (diff format) instead of a bespoke one.

Same comment for the JSON input.

Reply:
Nice idea, updated 👍🏾